cyrw/ioctl.go

```diff
@@ -327,6 +316,16 @@ func (d *Device) f2PacketAvail() (bool, uint16) {
 	return false, 0
 }
 
+// waitForCredit waits for a credit to use for the next transaction
+func (d *Device) waitForCredit(buf []uint32) error {
+	d.trace("waitForCredit")
+	if d.has_credit() {
+		return nil
+	}
+	_, err := d.pollForIoctl(buf) // credit is updated with poll.
+	return err
+}
```

Comment:
> This block or two of code gets repeated in several functions, in slightly diff ways. Can it be refactored into one func?

@scottfeldman  I've rewritten waitForCredit as a fix for your comment in other PR.

Reply:
Yeah I may have misinterpreted what pollForIoctl was doing- will refactor again taking this into consideration. Also thanks for the excellent explanation of how we could get pwned :sweat_smile: